src/main/java/io/gingersnapproject/cdc/configuration/Rule.java

```diff
@@ -15,12 +15,12 @@ public interface Rule {
       @WithDefault("|")
       String plainSeparator();
 
-      Optional<List<String>> keyColumns();
+      List<String> keyColumns();
 
       Optional<List<String>> columns();
```

Comment:
Should the columns be required?

Reply:
I don't think so. If none are explicitly configured  then we should return all columns in the table.

WDYT to changing the field name to `valueColumns` to make it's purpose more explicit?

Reply:
+1 for renaming, and also reflects the `keyColumns`.